Making prescription card status into a link. closes #3632

webinista opened the pull request:
Also:

- Modifies openGlossaryModal function in Detail.jsx so that glossary modal functions again. Closes #3657.
- Modifies behavior of GlossaryList to make the title optional.

webinista committed:
Making prescription card status into a link. closes #3632

Also:

- Modifies openGlossaryModal function in Detail.jsx so that glossary modal functions again. Closes #3657.
- Modifies behavior of GlossaryList to make the title optional.

webinista committed:
Making prescription card status into a link. closes #3632

Also:

- Modifies openGlossaryModal function in Detail.jsx so that glossary modal functions again. Closes #3657.
- Modifies behavior of GlossaryList to make the title optional.

U-DON commented on src/js/rx/containers/Detail.jsx:
```diff
@@ -165,7 +165,8 @@ export class Detail extends React.Component {
   openGlossaryModal(term) {
-    const content = glossary.filter((obj) => {
+    const allTerms = glossary.Prescription.concat(glossary.Refill);
+    const content = allTerms.filter((obj) => {
```
History needs the modal handling as well. Maybe add a helper that's shared between both.

U-DON commented on src/js/rx/components/Prescription.jsx:
```diff
@@ -92,8 +92,7 @@ class Prescription extends React.Component {
             className="rx-prescription-status">
-            Refill status:
-          <span> {rxStatuses[status]}</span>
+          Refill status: <Link to={`glossary#${status}`}>{rxStatuses[status]}</Link>
```
Should this pop up a modal like the other instances of glossary links? This would take the user directly to the glossary.

U-DON commented on src/js/rx/components/GlossaryList.jsx:
```diff
@@ -8,14 +8,20 @@ class GlossaryList extends React.Component {
     }
 
+    let title;
+    if (this.props.title) {
```
Add `title` to propTypes.

U-DON reviewed (changes requested):
There are also some merge conflicts.

webinista committed:
Adds 'title' as a GlossaryList prop type. Fixes merge conflicts.

webinista committed:
Makes refill status link open modal

- Adds a getGlossaryTerm helper
- Adds glossaryModalHandler, refillModalHandler props to Prescription.jsx, PrescriptionList.jsx

webinista committed:
Merge branch 'issue#3632' of github.com:department-of-veterans-affairs/vets-website into issue#3632

webinista committed:
Merge branch 'master' into issue#3632

webinista committed:
Merge branch 'master' into issue#3632

webinista committed:
Merge branch 'master' into issue#3632

webinista committed:
Merge branch 'issue#3632' of github.com:department-of-veterans-affairs/vets-website into issue#3632

U-DON committed:
Merge branch 'master' into issue#3632

U-DON committed:
Merge branch 'master' into issue#3632

U-DON commented on src/js/rx/components/GlossaryList.jsx:
```diff
@@ -8,14 +8,20 @@ class GlossaryList extends React.Component {
       if (terms.hasOwnProperty(o)) {
-        termsList.push(<dt key={++tKey}>{terms[o].term}</dt>);
+        let id = terms[o].term.toLowerCase().replace(/\s/g, '');
+        termsList.push(<dt id={id} key={++tKey}>{terms[o].term}</dt>);
```
Use `id` for the keys here too.

U-DON reviewed (changes requested):
History still needs to use the new glossary modal helper.

webinista committed:
Merge branch 'master' into issue#3632

webinista committed:
Re-adding the stray comma.

webinista committed:
Adding modal handler to History page.

webinista committed:
Merge branch 'master' into issue#3632

webinista committed:
Removing duplicate entry for 'Submitted' status. closes #3710

webinista committed:
Merge branch 'issue#3632' of github.com:department-of-veterans-affairs/vets-website into issue#3632

webinista committed:
Merge branch 'master' into issue#3632
